src/strategies/base.ts

```diff
@@ -254,30 +268,145 @@ export abstract class BaseStrategy implements Strategy {
+  async buildReleasePullRequest({
+    commits,
+    existingPullRequest,
+    labels = [],
+    latestRelease,
+    draft,
+    manifestPath,
+  }: {
+    commits: ConventionalCommit[];
+    latestRelease?: Release;
+    draft?: boolean;
+    labels?: string[];
+    existingPullRequest?: PullRequest;
+    manifestPath?: string;
+  }): Promise<ReleasePullRequest | undefined> {
     const conventionalCommits = await this.postProcessCommits(commits);
     this.logger.info(`Considering: ${conventionalCommits.length} commits`);
     if (conventionalCommits.length === 0) {
       this.logger.info(`No commits for path: ${this.path}, skipping`);
       return undefined;
     }
 
-    const newVersion = await this.buildNewVersion(
-      conventionalCommits,
-      latestRelease
+    const component = await this.getComponent();
+    this.logger.debug('component:', component);
+
+    const releaseAsCommit = conventionalCommits.find(conventionalCommit =>
+      conventionalCommit.notes.find(note => note.title === 'RELEASE AS')
+    );
+    const releaseAsNote = releaseAsCommit?.notes.find(
+      note => note.title === 'RELEASE AS'
     );
+
+    let newVersion: Version;
+    if (this.releaseAs) {
+      this.logger.warn(
+        `Setting version for ${this.path} from release-as configuration`
+      );
+      newVersion = Version.parse(this.releaseAs);
+    } else if (releaseAsNote) {
+      newVersion = Version.parse(releaseAsNote.text);
+    } else if (latestRelease) {
+      newVersion = await this.versioningStrategy.bump(
+        latestRelease.tag.version,
+        conventionalCommits
+      );
+    } else {
+      newVersion = this.initialReleaseVersion();
+    }
+
+    // If a pull request already exists, compare the manifest version from its branch against the one from the PR title.
+    // If they don't match, assume the PR title has been edited by an end user to set the version.
+    if (existingPullRequest) {
+      this.logger.info(
+        `PR already exists for ${existingPullRequest.headBranchName}, checking if PR title edited to set custom version`
+      );
+      const existingPRTitleVersion = PullRequestTitle.parse(
+        existingPullRequest.title
+      )?.getVersion();
+      const hasCustomVersionLabel = existingPullRequest.labels.find(
+        label => label === DEFAULT_CUSTOM_VERSION_LABEL
+      );
+
+      if (!existingPRTitleVersion && hasCustomVersionLabel) {
+        // report problem to end user
+        this.github.commentOnIssue(
+          `
+## Invalid version number in PR title
+
+:rotating_light: This Pull Request has the \`${DEFAULT_CUSTOM_VERSION_LABEL}\` label but the version number cannot be found in the title. Instead the generated version \`${newVersion}\` will be used.
+
+If you want to use a custom version be sure to use the semver format.
+
+If you do not want to set a custom version and want  to get rid of this warning, remove the label \`${DEFAULT_CUSTOM_VERSION_LABEL}\` from this Pull Request.
```

Comment:
Is the DEFAULT_CUSTOM_VERSION_LABEL set by default? i.e. the user has to remove a label they didn't add? Also, will removing the label remove this comment? If not, the user can't get rid of this warning

Reply:
Will this comment be repeated when the issue is not fixed?

Reply:
Yes, on new commits, or if the end user edits the title again then a new comment will be written.

Reply:
Ah, sorry, what I said is incorrect. I forgot a detail: if we cannot find the version in the title, then we fallback to the generated version based on conventional commits, which will update the PR title to something valid. So no, it won't repeat infinitely everytime release-please is triggered.

Reply:
I see. Ok, so the warning message should then say something about that no version was found in the title, so it was set to xxx.